packages/lab/src/dialog/DialogHeader.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  H2,
+  StatusIndicator,
+  Text,
+  type ValidationStatus,
+  makePrefixer,
+  useDialogContext,
+} from "@salt-ds/core";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import { useWindow } from "@salt-ds/window";
+import { clsx } from "clsx";
+import {
+  type ComponentPropsWithoutRef,
+  type ReactNode,
+  forwardRef,
+} from "react";
+import dialogHeaderCss from "./DialogHeader.css";
+
+const withBaseName = makePrefixer("saltDialogHeader");
+
+export interface DialogHeaderProps extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * The status of the Dialog
+   */
+  status?: ValidationStatus | undefined;
+  /**
+   * Displays the accent bar in the Dialog Title */
+  disableAccent?: boolean;
+  /**
+   * Displays the header at the top of the Dialog
+   */
+  header: ReactNode;
+  /**
+   * Displays the preheader just above the header
+   **/
+  preheader?: ReactNode;
+  /**
+   * Description text is displayed just below the header
+   **/
+  description?: ReactNode;
+  /**
+   * Actions to be displayed in header
+   */
+  actions?: ReactNode;
+}
+
+export const DialogHeader = forwardRef<HTMLDivElement, DialogHeaderProps>(
+  function DialogHeader(props, ref) {
+    const {
+      className,
+      description,
+      disableAccent,
+      actions,
+      header,
+      preheader,
+      status: statusProp,
+      ...rest
+    } = props;
+    const { status: statusContext, id } = useDialogContext();
+
+    const targetWindow = useWindow();
+    useComponentCssInjection({
+      testId: "salt-dialog-header",
+      css: dialogHeaderCss,
+      window: targetWindow,
+    });
+
+    const status = statusProp ?? statusContext;
+
+    return (
+      <div
+        id={id}
+        className={clsx(
+          withBaseName(),
+          {
+            [withBaseName("withAccent")]: !disableAccent && !status,
+            [withBaseName(status ?? "")]: !!status,
+          },
+          className,
+        )}
+        ref={ref}
+        {...rest}
+      >
+        {status && <StatusIndicator status={status} />}
+        <div className={withBaseName("container")}>
+          <div>
+            {preheader && (
+              <Text className={withBaseName("preheader")}>{preheader}</Text>
+            )}
+            <H2 className={withBaseName("header")}>{header}</H2>
```

Comment:
Similar heading hierarchy question

Reply:
updated the way we pass headers for both dialog and overlay headers so we don't constrain the components by wrapping a node in a header tag